BackEnd/images/views.py

```diff
@@ -1,86 +1,68 @@
-from rest_framework.generics import (
-    RetrieveUpdateAPIView,
-)
+from PIL import Image
+from hashlib import md5
+from rest_framework.generics import CreateAPIView, DestroyAPIView
 from rest_framework.parsers import MultiPartParser, FormParser
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
 from drf_spectacular.utils import extend_schema_view, extend_schema
 
-from profiles.permissions import UserIsProfileOwnerOrReadOnly
-from profiles.models import Profile
-from .serializers import BannerSerializer, LogoSerializer
-from utils.completeness_counter import completeness_count
+from profiles.permissions import UserIsImageOwner
+from images.models import ProfileImage
+from .serializers import ImageSerializer
 
 
 @extend_schema_view(
-    put=extend_schema(
+    post=extend_schema(
         request={
             "multipart/form-data": {
                 "type": "object",
                 "properties": {
-                    "banner_image": {"type": "string", "format": "binary"},
+                    "image_path": {"type": "string", "format": "binary"},
                 },
             }
         },
         responses={
-            200: BannerSerializer,
-        },
-    ),
-    patch=extend_schema(
-        request={
-            "multipart/form-data": {
-                "type": "object",
-                "properties": {
-                    "banner_image": {"type": "string", "format": "binary"},
-                },
-            }
-        },
-        responses={
-            200: BannerSerializer,
+            201: ImageSerializer,
         },
     ),
 )
-class BannerRetrieveUpdate(RetrieveUpdateAPIView):
-    permission_classes = (UserIsProfileOwnerOrReadOnly,)
-    serializer_class = BannerSerializer
+class ImageCreateAPIView(CreateAPIView):
+    permission_classes = [
+        IsAuthenticated,
+    ]
+    serializer_class = ImageSerializer
     parser_classes = (MultiPartParser, FormParser)
-    queryset = Profile.objects.active_only()
 
-    def perform_update(self, serializer):
-        completeness_count(serializer)
+    def get_serializer_context(self):
+        context = super().get_serializer_context()
+        image_type = self.kwargs.get("image_type")
+        context.update({"image_type": image_type})
+        return context
```

Comment:
Actually, there is no need to do this.

In Serializer code, you can do `self.context['view'].kwargs['image_type']`.

Reply:
Done